src/parser/parser.ts

```diff
@@ -1,8 +1,242 @@
+      const { start } = span;
+      const { line, character } = start;
+
+      switch (type) {
+        case TokenType.ID:
+          this.lexer.unlex(current);
+          const id = this.parseId();
+          args.push(id);
+          break;
+        case TokenType.LITERAL:
+          this.expect(current, TokenType.LITERAL, ",");
+          break;
+        default:
+          throw new ParserError(`${this.filename}: ${line}:${character}: unexpected token '${content}`);
+          break;
+      }
+      current = this.lexer.lexRegular();
+    }
+
+    const body = this.parseExpression();
+
+    return new Define(keyword, id, args, body);
+  }
+
+  private parseLet(): Expression {
+    const keyword = this.lexer.lexRegular();
+    this.expect(keyword, TokenType.LITERAL, "let");
+
+    const ids: Id[] = [];
+    const values: Expression[] = [];
+
+    const id0 = this.lexer.lexRegular();
+    this.expect(keyword, TokenType.ID);
+
+    const eq1 = this.lexer.lexRegular();
+    this.expect(eq1, TokenType.LITERAL, "=");
+
+    const v1 = this.parseExpression();
+
+    ids.push(new Id(id0));
+    values.push(v1);
+
+    let current = this.lexer.lexRegular();
+
+    while (current.content === ',') {
+      current = this.lexer.lexRegular();
+      this.expect(current, TokenType.ID);
+
+      const id = new Id(current);
+
+      current = this.lexer.lexRegular();
+      this.expect(current, TokenType.LITERAL, "=");
+
+      const value = this.parseExpression();
+
+      ids.push(id);
+      values.push(value);
+
+      this.lexer.lexRegular();
+    }
+
+    this.lexer.unlex(current);
+
+    const body = this.parseExpression();
+
+    return new Let(keyword, ids, values, body);
+  }
+
+  private parseIf(): Expression {
+    const keyword = this.lexer.lexRegular();
+    this.expect(keyword, TokenType.LITERAL, "if");
+
+    const predicate = this.parseExpression();
+
+    const thenToken = this.lexer.lexRegular();
+    this.expect(thenToken, TokenType.LITERAL, "then");
+
+    const consequent = this.parseExpression();
+
+    const elseToken = this.lexer.lexRegular();
+    this.expect(elseToken, TokenType.LITERAL, "else");
+
+    const alternative = this.parseExpression();
+
+    return new If(keyword, predicate, consequent, alternative);
+  }
+
+  private parseFor(): Expression {
+    const keyword = this.lexer.lexRegular();
+    this.expect(keyword, TokenType.LITERAL, "for");
+
+    const id = this.parseId();
+
+    const inToken = this.lexer.lexRegular();
+    this.expect(inToken, TokenType.LITERAL, "in");
+
+    const iterable = this.parseExpression();
+    const body = this.parseExpression();
+
+    return new For(keyword, id, iterable, body);
+  }
+
+  private parsePrint(): Expression {
+    const keyword = this.lexer.lexRegular();
+    this.expect(keyword, TokenType.LITERAL, "print");
+
+    const expression = this.parseExpression();
+
+    return new Print(keyword, expression);
+  }
+
+  // @ts-ignore
```

Comment:
Why's this needed?

Reply:
Oh, this is temporary, I'll take it out later - basically TSC is unhappy with the code as it is now